### PR TITLE
port: [#4465][#6560] Allow TokenCredential authentication in CosmosDbPartitionedStorage 

### PR DESCRIPTION
**Pull request by JhontSouth**
Fixes #minor

## Description
This PR adds the _CosmosDB_ authentication method through _TokenCredential_ in _CosmosDbPartitionedStorage_

## Specific Changes
- Upgraded _@azure/cosmos_ package from ^3.3.1 to 3.10.0 to get the _CosmosClient_ constructor that uses _TokenCredential_.
- Added _TokenCredential_ property to _CosmosDbPartitionedStorageOptions_ class.
- Updated validation to throw an error if _AuthKey_ and _TokenCredential_ properties are null.
- Added the alternative to create a _CosmosClient_ using the _TokenCredential_ if provided.
- Updated _Typescript_ version to avoid errors  with the new version of _@azure/cosmos_

## Testing
The following image shows the information saved in CosmosDB after creating the client using Azure credentials.
image

### CODE REVIEW

File: libraries/botbuilder-azure/package.json [hunk 1]
```diff
@@ -56,4 +57,4 @@
     "lib",
     "src"
   ]
-}
+}
```
Comment (sw-joelmut) [inline libraries/botbuilder-azure/package.json:60]: Add the end of line (EOF).

File: package.json [hunk 2]
```diff
@@ -110,4 +110,4 @@
       "**/internal.*"
     ]
   }
-}
+}
```
Comment (sw-joelmut) [inline package.json:113]: Add the end of line (EOF).

**Review by sw-joelmut — CHANGES_REQUESTED** [review]
I think the `err: any` are ok, another solution could have been supporting the `unknown` type in most cases like `Error | unknown`, but it's ok.

**Review by ceciliaavila — CHANGES_REQUESTED** [review]
Add the issue number in the PR title and in the description remove the `^` from the new cosmos package version.

**Review by ceciliaavila — APPROVED** [review]

**Review by sw-joelmut — APPROVED** [review]
Just a reminder that when we promote the PR add the issue number in the Fixes # xxx description

**Pull request closed without merging**